### PR TITLE
[docs] Update the contribution guide

### PR DESCRIPTION
**Pull request by DIRECTcut**
is part of [DevExpress/testcafe#5195](https://github.com/DevExpress/testcafe/issues/5195)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -30,34 +30,73 @@ Follow the steps below when submitting your code.
...
+    node node_modules/gulp/bin/gulp http-playground
+    ```
 
-    Add unit and/or functional tests if you are developing a new functionality.
+    The `http-playground` task builds Hammerhead, launches a local HTTP server, and opens a playground page, where you can specify a web page to proxy.
+
+    To run a playground over HTTPS, run the `https-playground` task.
+
+    ```sh
+    node node_modules/gulp/bin/gulp https-playground
```
Comment (DIRECTcut) [inline CONTRIBUTING.md:65]: Do we need to launch gulp directly from `bin`? Can't we do this instead:

```sh
 node gulp https-playground
 ```

Comment (AlexSkorkin) [reply CONTRIBUTING.md:65]: You can make it even simpler:

`gulp https-playground`

See step 9 -- there's a short command used.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,34 +30,73 @@ Follow the steps below when submitting your code.
...
     ```
 
-10. Push changes to your fork.
+10. Push changes to your fork and open a pull request.
+
+Before you submit your pull request, it has to satisfy the following conditions:
+
+* The pull request name should describe the changes you implemented.
+* The pull request description should contain the [closes](https://github.com/blog/1506-closing-issues-via-pull-requests) directive with an appropriate issue number.
+* All tests must be passing
```
Comment (AlexSkorkin) [inline CONTRIBUTING.md:87]:
```suggestion
* All existing and new tests must be passing
```

Probably worth clarifying what tests must pass

**Review by AlexSkorkin — COMMENTED** [review]
Looks good, you might want to address comments too.

**Review by VasilyStrelyaev — APPROVED** [review]

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -19,8 +19,8 @@ Join the TestCafe Hammerhead community on Stack Overflow: ask and answer [questi
 
 ## Reporting a Problem
 
-If you encounter a bug when using TestCafe Hammerhead, please file an issue in our [GitHub repository](https://github.com/DevExpress/testcafe-hammerhead/issues).
-We recommend searching through the existing issues to see if the problem has already been reported or addressed.
+If you run into a bug with TestCafe Hammerhead, please file an issue in the [GitHub repository](https://github.com/DevExpress/testcafe-hammerhead/issues).
```
Comment (albertov05) [inline CONTRIBUTING.md:22]:
```suggestion
If you encounter a bug with TestCafe Hammerhead, please file an issue in the [GitHub repository](https://github.com/DevExpress/testcafe-hammerhead/issues).
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -19,8 +19,8 @@ Join the TestCafe Hammerhead community on Stack Overflow: ask and answer [questi
...
+Search through the existing issues to see if the problem has already been reported or addressed.
 
 When you create a new issue, the template text is automatically added to its body. You should complete all the sections in this template to help us understand the issue you are describing. Missing information could delay the processing time.
```
Comment (albertov05) [inline CONTRIBUTING.md:25]:
```suggestion
When you create a new issue, the template text is automatically added to its body. You should complete all sections in this template to help us understand the issue you are describing. Missing information could delay the processing time.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -30,34 +30,73 @@ Follow the steps below when submitting your code.
 
 1. Search the [list of issues](https://github.com/DevExpress/testcafe-hammerhead/issues) to see if there is an issue for the bug or feature you are going to work on or create a new one.
 
-2. If you are going to address an existing issue, check the comment thread to make sure that nobody is working on it at the moment.
+2. To address an already described issue, check the comment thread to make sure that nobody is working on it at the moment. Leave a comment saying that you are willing to fix this issue, and include details on how you plan to do this. Core team members may need to discuss the details of the proposed fix with you. After the green light from them,
```
Comment (albertov05) [inline CONTRIBUTING.md:33]:
```suggestion
2. To address an already described issue, check the comment thread to make sure that no one is working on it at the moment. Leave a comment saying that you are willing to fix this issue, and include details on how you plan to do this. Core team members may need to discuss the details of the proposed fix with you. After they have approved it,
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -30,34 +30,73 @@ Follow the steps below when submitting your code.
...
+leave a comment saying that you started your work on this issue.
```
Comment (albertov05) [inline CONTRIBUTING.md:34]: Just out of curiousity, why is the last part of line 33 on this line instead?

Comment (DIRECTcut) [reply CONTRIBUTING.md]: That's a mistake. Good that you've noticed, thanks.

**Pull request merged**